src/hardware/hardware.cpp

```diff
@@ -533,19 +533,23 @@ void CAPTURE_AddImage(Bitu width, Bitu height, Bitu bpp, Bitu pitch, Bitu flags,
 				switch ( bpp) {
 				case 8:
 					for (x=0;x<countWidth;x++)
-						((Bit8u *)doubleRow)[x*2+0] =
-						((Bit8u *)doubleRow)[x*2+1] = ((Bit8u *)srcLine)[x];
+						doubleRow[x*2+0] =
+						doubleRow[x*2+1] = srcLine[x];
 					break;
 				case 15:
 				case 16:
-					for (x=0;x<countWidth;x++)
-						((Bit16u *)doubleRow)[x*2+0] =
-						((Bit16u *)doubleRow)[x*2+1] = ((Bit16u *)srcLine)[x];
+					for (x = 0; x < countWidth; x++) {
+						const uint16_t pixel = host_readw_at(srcLine, x);
+						host_writew_at(doubleRow, x * 2, pixel);
+						host_writew_at(doubleRow, x * 2 + 1, pixel);
+					}
 					break;
 				case 32:
-					for (x=0;x<countWidth;x++)
-						((Bit32u *)doubleRow)[x*2+0] =
-						((Bit32u *)doubleRow)[x*2+1] = ((Bit32u *)srcLine)[x];
```

Comment:
PPC: Fix needs to be aligned and native endian (not swapped to little-endian like the host_* functions do).